### PR TITLE
Allow using between 1 and 8 cascades for SDFGI

### PR DESCRIPTION
**Pull request by Calinou**
Related to https://github.com/godotengine/godot/pull/39965 (can be merged independently).

This provides more flexibility between performance and quality adjustments, especially when using SDFGI for small-scale levels (which can be useful for real-time procedurally generated scenes).

**Testing project:** [test_sdfgi_num_cascades.zip](https://github.com/godotengine/godot/files/7865266/test_sdfgi_num_cascades.zip)

## Preview

*Look at the FPS counter in the top-right corner.*

### 1 cascade

*Since GI would end really up close with a cell size of 0.4, I manually increased the cell size to 0.6.*

### 1 cascade + half-resolution GI + 8 SDFGI rays

*The lowest-end possible setting. Only takes 0.5 ms of GPU time on a GTX 1080. Should be usable on recent or high-end integrated graphics.*

### CODE REVIEW

**Comment by mrjustaguy** [comment]
This is really great, 4 cascades Stuttered on Low end GPUs, while 1-2 don't. Thanks for this change!

**Comment by clayjohn** [comment]
Looks good to me too! I'm surprised how good the quality is with just one cascade. How does it look in motion?

**Comment by mrjustaguy** [comment]
Just fine, same as static really... now with pair this with 1 frame for converging and it could be a fun effect.. I mean it isn't a very strong "GI" effect that way, but it is an interesting and good looking effect that way.

It would be interesting to see how this would work if you could increase the size of a Cascade without affecting cell size..

**Comment by Calinou** [comment]
> How does it look in motion?

In motion, the fade-in is quite noticeable indoors by default (but it's OK outdoors):

https://user-images.githubusercontent.com/180032/149562178-7c63217c-f28a-4bec-a513-f3c2dc0b9751.mp4

However, you can improve the appearance of indoor SDFGI fade-in (at the cost of outdoor appearance) by exploiting the fact that [Environment ambient light settings can be adjusted independently of SDFGI ambient light settings](https://github.com/godotengine/godot/issues/55104).

Here's an example with Sky Contribution decreased to 0.25. Indoor fade-in becomes much less noticeable, but outdoor appearance is worse:

https://user-images.githubusercontent.com/180032/149562285-0be4a33f-f282-48be-9bb3-83a8480ab8ab.mp4

**Comment by clayjohn** [comment]
Looks much better than having no GI to me! Even in motion!

File: doc/classes/Environment.xml [hunk 1]
```diff
@@ -168,7 +168,8 @@
 		</member>
 		<member name="sdfgi_cascade0_distance" type="float" setter="set_sdfgi_cascade0_distance" getter="get_sdfgi_cascade0_distance" default="12.8">
 		</member>
-		<member name="sdfgi_cascades" type="int" setter="set_sdfgi_cascades" getter="get_sdfgi_cascades" enum="Environment.SDFGICascades" default="1">
+		<member name="sdfgi_cascades" type="int" setter="set_sdfgi_cascades" getter="get_sdfgi_cascades" default="6">
+			The number of cascades to use for SDFGI. A higher number of cascades allows displaying SDFGI further away while preserving detail up close, at the cost of performance. When using SDFGI on small-scale levels, [member sdfgi_cascades] can often be decreased between [code]1[/code] and [code]4[/code] to improve performance.
```
Comment (akien-mga) [inline doc/classes/Environment.xml:172]: This should mention that possible values range from 1 to 8.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**